### PR TITLE
Capfile fixes

### PR DESCRIPTION
**Pull request by orenmazor**
bunch of changes I've been cherry-picking around all day, and a bug fix.

@Shopify/data-engineers

### CODE REVIEW

File: Capfile [hunk 1]
```diff
@@ -12,7 +12,7 @@ set :keep_releases, 5
 
 DATANODES = (2..47).map {|i| "dn%02d.chi.shopify.com" % i }
 OTHERNODES = ["hadoop-etl1.chi.shopify.com", "spark-etl1.chi.shopify.com", "reports-reportify-etl3.chi.shopify.com", "reports-reportify-skydb4.chi.shopify.com", "platfora2.chi.shopify.com"]
-BROKEN = ["dn16.chi.shopify.com"] # Node is down don't try to send code
+BROKEN = ["dn09.chi.shopify.com", "dn16.chi.shopify.com"] # Node is down don't try to send code
```
Comment (orenmazor) [inline Capfile]: dn09 is not available. eventually I'd like to integrated this with CDM's API to just check which nodes are not available and force the user to accept that nodes not-deployability manually.

File: Capfile [hunk 2]
```diff
@@ -34,13 +34,19 @@ namespace :deploy do
     existing_releases.reject! {|element| element.end_with?("spark-assembly-latest.jar")}
     if existing_releases.count > count
       existing_releases.shift(existing_releases.count - count).each do |path|
-        run "hdfs dfs -rm -skipTrash #{path}"
+        run "hdfs dfs -rm #{path}"
```
Comment (orenmazor) [inline Capfile]: this was dumb. I was deleting jars and throwing them away immediately. in the interim, setting this to at least save to trash (~30 days IIRC)

File: Capfile [hunk 3]
```diff
@@ -34,13 +34,19 @@ namespace :deploy do
...
       end
     end
   end
 
   task :upload_to_hdfs, :roles => :uploader, :on_no_matching_servers => :continue do
```
Comment (orenmazor) [inline Capfile]: fetch(:sha) was failing for me. I am going to look into this, actually and not merge my hack here.

File: Capfile [hunk 4]
```diff
@@ -34,13 +34,19 @@ namespace :deploy do
...
-    run "hdfs dfs -copyFromLocal -f /u/apps/spark/current/lib/spark-assembly-*.jar #{fetch(:spark_jar_path)}/spark-assembly-#{fetch(:sha)}.jar"
+    target_sha = ENV['SHA'] || `git rev-parse HEAD`.gsub(/\s/,"")
+    run "hdfs dfs -copyFromLocal -f /u/apps/spark/current/lib/spark-assembly-*.jar hdfs://nn01.chi.shopify.com/user/sparkles/spark-assembly-#{target_sha}.jar"
+  end
+
+  task :litmus_test_sha_jar, :roles => :uploader, :on_no_master_servers => :continue do
```
Comment (orenmazor) [inline Capfile]: this just runs canary from the driver to make sure the jar is loadable and runnable.

Comment (udnay) [inline Capfile]: What happens if the jar is busted, will the deploy fail?

Comment (orenmazor) [inline Capfile]: eventually i'd like that to be the case. right now this is not going to be the case.

Comment (udnay) [inline Capfile]: Cool, this is better than having to remember to do it manually. I wonder if we could do a special flow `deploy_canary` for this which causes a page or something. If the deploy goes sideways then we want to know asap, but probably out of the scope of this PR.

**Comment by udnay** [comment]
Other than questions LGTM

File: Capfile (hunk 3 above)
Comment (orenmazor) [inline Capfile]: actually the sha variable is set by capistrano further down the call stack, but it doesnt work if you just call upload_to_hdfs or test_spark_jar independently.

**Pull request merged**